modules/tinymce/src/core/main/ts/file/Uploader.ts

```diff
@@ -36,8 +36,9 @@ export interface UploadFailureOptions {
   remove?: boolean;
 }
 
-export type UploadHandler = (blobInfo: BlobInfo, success: (url: string) => void, failure: (err: string, options?: UploadFailureOptions) => void, progress?: (percent: number) => void) => void;
+export type UploadHandler = (blobInfo: BlobInfo, success: (url: string) => void, failure: (err: string, options?: UploadFailureOptions) => void, progress: ProgressFn) => void;
 
+type ProgressFn = (percent: number) => void;
```

Comment:
Can we please put this above the `UploadHandler` type? As I'm not a fan of having types (or variables for that matter) used before they are defined.

Reply:
Fixed